cqrs-codegen/impl/src/event/typed_event.rs

```diff
@@ -96,35 +96,43 @@ fn derive_enum(input: syn::DeriveInput) -> Result<TokenStream> {
 
     let (impl_generics, ty_generics, _) = input.generics.split_for_impl();
 
-    // TODO: Remove this once generics are supported in const contexts.
-    //       https://github.com/instrumentisto/cqrs-rs/pull/21#issuecomment-1257982811
-    let fake_params = input.generics.params.iter().filter_map(|p| match p {
-        syn::GenericParam::Const(p) => {
-            let ident = &p.ident;
-            let ty = &p.ty;
-            // We can use `0` here, because only numeric types are allowed
-            // as const generics.
-            Some(quote! { const #ident: #ty = 0; })
-        },
-        syn::GenericParam::Lifetime(_) => None,
-        syn::GenericParam::Type(p) => {
-            let ident = &p.ident;
-            Some(quote! { type #ident = (); })
-        },
-    });
-
     let subtypes = types
         .iter()
-        .map(|ty| quote!(<#ty as ::cqrs::TypedEvent>::EVENT_TYPES))
+        .map(|ty| quote! { <#ty as ::cqrs::TypedEvent>::EVENT_TYPES })
         .collect::<Vec<_>>();
 
+    let len = quote! {
+        0 #(+ #subtypes.len())*
+    };
+
     Ok(quote! {
         #[automatically_derived]
         impl#impl_generics ::cqrs::TypedEvent for #type_name#ty_generics #where_clause {
             #[doc = #const_doc]
             const EVENT_TYPES: &'static [::cqrs::EventType] = {
-                #( #fake_params )*
-                ::cqrs::const_concat_slices!(::cqrs::EventType, #( #subtypes ),*)
+                ::cqrs::private::slice_arr(
```

Comment:
@50U10FCA7 this outer block could be removed and we can leave only `const EVENT_TYPES: &'static [::cqrs::EventType] = ::cqrs::private::slice_arr(`